components/Chat/ChatInput.vue

```diff
@@ -20,22 +20,25 @@
         autocapitalize="off"
         spellcheck="true"
         single-line
+        auto-grow
         validate-on-blur
         outlined
         dense
         clearable
+        rows="1"
         counter="300"
         @focus="onFocus"
         @change="value => this.setChatMessage( value )"
         @keyup.delete="updateMessage"
-        @keyup.enter.prevent="sendMessage"
```

Comment:
Why was keyup modified to keydown? This appears to cause behavioral issues due to timing and the race condition of setting the value in the store the event firing off.
Keydown appears to fire too early and before the store may have updated (e.g. in the instance of text pasted into an empty text field, upon hitting enter, reports the field value as empty).

Reply:
Hm, I thought I'd checked for this. Keydown seems to clash with `v-textarea`'s return key handler, which leads to a \n being appended to the message, sometimes not sending it completely. Perhaps force an update on keydown?

Reply:
I think event handling might need to be re-worked to work properly with `text-area` since it's relying on what seems like race-conditions on `text-field` currently.
I'm going to revert the chat input modifications for now, and we should start a new PR with the input modifications to discuss proper event handling for text area.

This component has the major downside of having many edge cases.